src/Database/Query.php

```diff
@@ -132,10 +133,10 @@ class Query implements ExpressionInterface, IteratorAggregate
     /**
      * Constructor.
      *
-     * @param \Cake\Datasource\ConnectionInterface $connection The connection
+     * @param \Cake\Database\ConnectionInterface $connection The connection
      * object to be used for transforming and executing this query
      */
-    public function __construct($connection)
+    public function __construct(ConnectionInterface $connection)
```

Comment:
This new typehint might cause issues if there are useland extensions of the Query object.